stage1/init/common/pod.go

```diff
@@ -565,6 +565,56 @@ func PodToSystemd(p *stage1commontypes.Pod, interactive bool, flavor string, pri
 	return nil
 }
 
+// evaluateAppMountPath tries to resolve symlinks within the path.
+// It returns the actual relative path for the given path.
+// TODO(yifan): This is a temporary fix for systemd-nspawn not handling symlink mounts well.
+// Could be removed when https://github.com/systemd/systemd/issues/2860 is resolved, and systemd
+// version is bumped.
+func evaluateAppMountPath(appRootfs, path string) (string, error) {
+	link := appRootfs
+
+	paths := strings.Split(path, "/")
+	for i, p := range paths {
+		next := filepath.Join(link, p)
+
+		if !strings.HasPrefix(next, appRootfs) {
+			return "", fmt.Errorf("path escapes app's root: %q", path)
```

Comment:
Instead of returning an error, it could just continue with appRootfs:
```
if !strings.HasPrefix(next, appRootfs) {
  link = appRootfs
  continue
}
```

So if `path` is "../../../../../../usr", this would just return "usr" without escaping the appRootfs.